File: src/main/rooms/Room.java
package main.rooms;

import main.Entity;
import main.Language;
import processing.core.PConstants;
import processing.core.PImage;

public abstract class Room extends Entity {

    void drawTitle(String section, String key) {
        main.noStroke();
        main.fill(0);
        main.rect(0, 0, main.screenWidth, main.screenHeight);

        main.textSize(40);
        main.textAlign(PConstants.CENTER, PConstants.CENTER);
        main.fill(255);
        main.text(Language.getLanguageSelected().getLocalizedString(section, key), main.screenWidth / 2F, 60);
    }

    public abstract void setup();

    public abstract void draw();

    public void end() {

    }

    public void mousePressed() {

    }

    public void keyPressed() {

    }

}

File: src/main/rooms/Room.kt
package main.rooms

import main.Language
import main.Main
import processing.core.PConstants
import processing.core.PImage

abstract class Room {
    internal val plus: PImage = Main.loadImage("resources/images/plus.png")
    internal val equal: PImage = Main.loadImage("resources/images/equal.png")

    fun drawTitle(section: String, key: String) {
        Main.textSize(40f)
        Main.textAlign(PConstants.CENTER, PConstants.CENTER)
        Main.fill(255)
        Main.text(Language.languageSelected.getLocalizedString(section, key), Main.screenWidth / 2f, 60f)
    }

    abstract fun setup()
    abstract fun draw()
    open fun end() {}
    open fun mousePressed() {}
    open fun keyPressed() {}
}